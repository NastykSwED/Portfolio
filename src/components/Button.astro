---
import { Icon } from 'astro-icon';

interface Props {
	text: string;
	variant?: string;
	type?: 'button' | 'submit' | 'reset' | undefined;
	iconName: string;
	iconPackName: string;
	iconUtility?: string;
}
const { text, type, variant, iconPackName, iconName, iconUtility } =
	Astro.props;
---

<button class={variant ? `Button Button--${variant}` : 'Button'} type={type}
	>{text}
	<span class={`Button-Icon ${iconUtility}`}>
		<Icon pack={iconPackName} name={iconName} class={iconUtility} />
	</span>
</button>
