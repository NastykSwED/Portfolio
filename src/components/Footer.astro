---
const year = new Date().getFullYear();
---

<footer class='re-Footer' id='re-Footer'>
	<div class='re-Footer-head'>
		<div class='re-Footer-content'>
			<h1 class='re-Footer-title'>Ronald Torres</h1>
			<h2 class='re-Footer-subTitle'>Front End Developer</h2>
		</div>
		<div class='re-Footer-enjoyer'>
			<span>React Enjoyer</span>
		</div>
	</div>
	<div class='re-Footer-legal'>
		<div class='re-Footer-disclaimer'>
			<p class='re-Footer-disclaimerTitle'>
				Â©{year} Ronald Torres. All rights reserved
			</p>
		</div>
	</div>
</footer>
