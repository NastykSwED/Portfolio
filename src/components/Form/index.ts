export * from './ContactFormField';
export * from './ContactFormInput';
export * from './ContactFormTextArea';
