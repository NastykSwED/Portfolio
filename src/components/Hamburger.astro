<div class='re-Hamburger'>
	<span class='re-Hamburger-bar'></span>
	<span class='re-Hamburger-bar'></span>
	<span class='re-Hamburger-bar'></span>
</div>

<script>
	const hamburgerElement = document.querySelector(
		'.re-Hamburger'
	) as HTMLElement;
	const headerNavbarElement = document.querySelector(
		'.re-Navbar'
	) as HTMLElement;

	const handleClick = () => {
		hamburgerElement.classList.toggle('isActive');
		headerNavbarElement.classList.toggle('isActive');
	};

	hamburgerElement.addEventListener('click', handleClick);
</script>
