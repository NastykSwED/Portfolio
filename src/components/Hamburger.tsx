import { useState } from 'react';

interface HamburgerProps {
	onClick: () => void;
}

export const Hamburger = (props: HamburgerProps) => {
	const { onClick } = props;

	const [isActive, setIsActive] = useState<boolean>(false);

	return (
		<div
			data-testid='Hamburger'
			onClick={() => {
				onClick();
				setIsActive(!isActive);
			}}
			className={`re-Hamburger ${isActive ? 'isActive' : ''}`}
		>
			<span className='re-Hamburger-bar'></span>
			<span className='re-Hamburger-bar'></span>
			<span className='re-Hamburger-bar'></span>
		</div>
	);
};
