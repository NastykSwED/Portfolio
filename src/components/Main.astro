---
import ContactSection from './Sections/Contact/ContactSection.astro';
import HomeSection from './Sections/Home/HomeSection.astro';
import ProjectsSection from './Sections/Projects/ProjectsSection.astro';
import SkillsSection from './Sections/Skills/SkillsSection.astro';
---

<main class='re-Main' id='re-Main'>
	<HomeSection />
	<SkillsSection />
	<ProjectsSection />
	<ContactSection />
</main>
