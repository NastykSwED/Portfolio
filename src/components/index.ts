export * from './Buttons';
export * from './Form';
export * from './Loaders';
export * from './Section';
export * from './Navbar/Navbar';
export * from './Hamburger';

