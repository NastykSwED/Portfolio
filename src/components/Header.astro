---
import Hamburger from './Hamburger.astro';
import Navbar from './Navbar/Navbar.astro';
---

<header class='re-Header' id='re-Header'>
	<div class='re-Header-wrapNavbar'>
		<h1 class='re-Header-logo'>Ronald Torres</h1>
		<Navbar />
		<Hamburger />
	</div>
</header>
