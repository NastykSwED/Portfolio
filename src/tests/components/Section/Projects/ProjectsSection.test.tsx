import { describe, it, expect, vi } from 'vitest';

import { render } from '@testing-library/react';

import { ProjectsSection } from '../../../../components';

describe('Testing ProjectsSection component', () => {
	it('Should match the snapshot', () => {
		const { container } = render(<ProjectsSection />);

		expect(container).toMatchSnapshot();
	});
});
