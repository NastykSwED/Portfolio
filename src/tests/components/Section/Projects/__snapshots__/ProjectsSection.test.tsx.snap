// Vitest Snapshot v1

exports[`Testing ProjectsSection component > Should match the snapshot 1`] = `
<div>
  <section
    class="re-Main-ProjectsSection"
    id="re-ProjectsSection"
  >
    <div
      class="re-ProjectsSection-content"
    >
      <h1
        class="re-ProjectsSection-title"
      >
        Projects
      </h1>
      <h2
        class="re-ProjectsSection-subtitle"
      >
        Most recent work
      </h2>
    </div>
    <div
      class="re-ProjectsSection-wrapProjects"
    >
      <div
        class="re-ProjectsSection-wrapProjectCard"
      >
        <div
          class="re-ProjectCard"
        >
          <div
            class="re-ProjectCard-wrapImage"
          >
            <img
              alt="Journal App Logo"
              class="re-ProjectCard-image"
              src="/assets/projects/journal-app.webp"
            />
          </div>
          <div
            class="re-ProjectCard-content"
          >
            <div
              class="re-ProjectCard-wrapBody"
            >
              <h1
                class="re-ProjectCard-Title"
              >
                Journal App
              </h1>
              <div
                class="re-ProjectCard-wrapLinks"
              >
                <a
                  aria-label="Find out more information about the project through this link"
                  class="re-ProjectCard-link"
                  href="https://github.com/NastykSwED/journal-app"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <span
                    class="re-ProjectCard-wrapIcon"
                  >
                    <svg
                      class="Icon Icon--FiGithub"
                      fill="none"
                      height="1em"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 0 0-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0 0 20 4.77 5.07 5.07 0 0 0 19.91 1S18.73.65 16 2.48a13.38 13.38 0 0 0-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 0 0 5 4.77a5.44 5.44 0 0 0-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 0 0 9 18.13V22"
                      />
                    </svg>
                  </span>
                </a>
                <a
                  aria-label="Find out more information about the project through this link"
                  class="re-ProjectCard-link"
                  href="https://journal-app-nastykswed.netlify.app/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <span
                    class="re-ProjectCard-wrapIcon"
                  >
                    <svg
                      class="Icon Icon--IoOpenOutline"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M384 224v184a40 40 0 01-40 40H104a40 40 0 01-40-40V168a40 40 0 0140-40h167.48M336 64h112v112M224 288L440 72"
                        fill="none"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="32"
                      />
                    </svg>
                  </span>
                </a>
              </div>
            </div>
            <p
              class="re-ProjectCard-Description"
            >
              Keep track of what happens in your day to day life with this application.
            </p>
            <div
              class="re-ProjectCard-wrapTools"
            >
              <span
                class="re-ProjectCard-tool"
              >
                TypeScript
              </span>
              <span
                class="re-ProjectCard-tool"
              >
                React Js
              </span>
              <span
                class="re-ProjectCard-tool"
              >
                Redux
              </span>
              <span
                class="re-ProjectCard-tool"
              >
                Material UI
              </span>
              <span
                class="re-ProjectCard-tool"
              >
                Firebase
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="re-ProjectsSection-wrapProjectCard"
      >
        <div
          class="re-ProjectCard"
        >
          <div
            class="re-ProjectCard-wrapImage"
          >
            <img
              alt="Todo App Logo"
              class="re-ProjectCard-image"
              src="/assets/projects/todo-app.webp"
            />
          </div>
          <div
            class="re-ProjectCard-content"
          >
            <div
              class="re-ProjectCard-wrapBody"
            >
              <h1
                class="re-ProjectCard-Title"
              >
                Todo App
              </h1>
              <div
                class="re-ProjectCard-wrapLinks"
              >
                <a
                  aria-label="Find out more information about the project through this link"
                  class="re-ProjectCard-link"
                  href="https://github.com/NastykSwED/todo-app"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <span
                    class="re-ProjectCard-wrapIcon"
                  >
                    <svg
                      class="Icon Icon--FiGithub"
                      fill="none"
                      height="1em"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 0 0-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0 0 20 4.77 5.07 5.07 0 0 0 19.91 1S18.73.65 16 2.48a13.38 13.38 0 0 0-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 0 0 5 4.77a5.44 5.44 0 0 0-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 0 0 9 18.13V22"
                      />
                    </svg>
                  </span>
                </a>
                <a
                  aria-label="Find out more information about the project through this link"
                  class="re-ProjectCard-link"
                  href="https://todo-app-nastykswed.netlify.app/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <span
                    class="re-ProjectCard-wrapIcon"
                  >
                    <svg
                      class="Icon Icon--IoOpenOutline"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M384 224v184a40 40 0 01-40 40H104a40 40 0 01-40-40V168a40 40 0 0140-40h167.48M336 64h112v112M224 288L440 72"
                        fill="none"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="32"
                      />
                    </svg>
                  </span>
                </a>
              </div>
            </div>
            <p
              class="re-ProjectCard-Description"
            >
              Simple Todo App using React JS and Tailwind CSS
            </p>
            <div
              class="re-ProjectCard-wrapTools"
            >
              <span
                class="re-ProjectCard-tool"
              >
                TypeScript
              </span>
              <span
                class="re-ProjectCard-tool"
              >
                React JS
              </span>
              <span
                class="re-ProjectCard-tool"
              >
                Tailwind CSS
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="re-ProjectsSection-wrapProjectCard"
      >
        <div
          class="re-ProjectCard"
        >
          <div
            class="re-ProjectCard-wrapImage"
          >
            <img
              alt="Click Counter Logo"
              class="re-ProjectCard-image"
              src="/assets/projects/click-counter.webp"
            />
          </div>
          <div
            class="re-ProjectCard-content"
          >
            <div
              class="re-ProjectCard-wrapBody"
            >
              <h1
                class="re-ProjectCard-Title"
              >
                Click Counter
              </h1>
              <div
                class="re-ProjectCard-wrapLinks"
              >
                <a
                  aria-label="Find out more information about the project through this link"
                  class="re-ProjectCard-link"
                  href="https://github.com/NastykSwED/click-counter"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <span
                    class="re-ProjectCard-wrapIcon"
                  >
                    <svg
                      class="Icon Icon--FiGithub"
                      fill="none"
                      height="1em"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 0 0-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0 0 20 4.77 5.07 5.07 0 0 0 19.91 1S18.73.65 16 2.48a13.38 13.38 0 0 0-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 0 0 5 4.77a5.44 5.44 0 0 0-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 0 0 9 18.13V22"
                      />
                    </svg>
                  </span>
                </a>
                <a
                  aria-label="Find out more information about the project through this link"
                  class="re-ProjectCard-link"
                  href="https://nastykswed.github.io/click-counter/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <span
                    class="re-ProjectCard-wrapIcon"
                  >
                    <svg
                      class="Icon Icon--IoOpenOutline"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M384 224v184a40 40 0 01-40 40H104a40 40 0 01-40-40V168a40 40 0 0140-40h167.48M336 64h112v112M224 288L440 72"
                        fill="none"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="32"
                      />
                    </svg>
                  </span>
                </a>
              </div>
            </div>
            <p
              class="re-ProjectCard-Description"
            >
              Simple Click Counter App using React JS
            </p>
            <div
              class="re-ProjectCard-wrapTools"
            >
              <span
                class="re-ProjectCard-tool"
              >
                React Js
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
